crates/bitcoin-da/src/service.rs

```diff
@@ -312,52 +354,135 @@ impl BitcoinService {
-            fee_sat_per_vbyte,
-            network,
-            self.reveal_tx_id_prefix.as_slice(),
-        )?;
-
-        // sign inscribe transactions
-        // let serialized_unsigned_commit_tx = &encode::serialize(&unsigned_commit_tx);
-        let signed_raw_commit_tx = client
-            .sign_raw_transaction_with_wallet(&unsigned_commit_tx, None, None)
-            .await?;
-
-        // send inscribe transactions
-        client
-            .send_raw_transaction(&signed_raw_commit_tx.hex)
-            .await?;
-
-        // serialize reveal tx
-        let serialized_reveal_tx = &encode::serialize(&reveal_tx.tx);
+        match da_data {
+            DaData::ZKProof(_) => {
+                // create inscribe transactions
+                let inscription_txs = create_zkproof_transactions(
+                    &rollup_name,
+                    blob,
+                    &da_private_key,
+                    prev_utxo,
+                    utxos,
+                    address,
+                    REVEAL_OUTPUT_AMOUNT,
+                    fee_sat_per_vbyte,
+                    fee_sat_per_vbyte,
+                    network,
+                    &self.reveal_light_client_prefix,
+                )?;
+
+                // write txs to file, it can be used to continue revealing blob if something goes wrong
+                write_inscription_txs(&inscription_txs);
+
+                match inscription_txs {
+                    LightClientTxs::Complete { commit, reveal } => {
+                        // sign inscribe transactions
+                        let signed_raw_commit_tx = client
+                            .sign_raw_transaction_with_wallet(&commit, None, None)
+                            .await?;
+
+                        // send inscribe transactions
+                        client
+                            .send_raw_transaction(&signed_raw_commit_tx.hex)
+                            .await?;
+
+                        // serialize reveal tx
+                        let serialized_reveal_tx = &encode::serialize(&reveal.tx);
+
+                        // send reveal tx
+                        let reveal_tx_hash =
+                            client.send_raw_transaction(serialized_reveal_tx).await?;
+
+                        info!("Blob inscribe tx sent. Hash: {}", reveal_tx_hash);
+                        Ok(reveal)
+                    }
+                    LightClientTxs::Chunked {
+                        commit_chunks,
+                        reveal_chunks,
+                        commit,
+                        reveal,
+                    } => {
+                        for (commit, reveal) in commit_chunks.into_iter().zip(reveal_chunks) {
+                            // sign inscribe transactions
+                            let signed_raw_commit_tx = client
+                                .sign_raw_transaction_with_wallet(&commit, None, None)
+                                .await?;
+
```

Comment:
can you also write the txs in a file just like we do on seq. comm.

Reply:
I do with write_inscription_txs above